addons/mail/static/src/messaging/entity_field.js

```diff
@@ -428,9 +428,16 @@ class EntityField {
             const newVal = [];
             for (const otherEntity of entity[relationName]) {
                 const otherField = otherEntity.constructor.fields[relatedFieldName];
-                const v = otherField.get(otherEntity);
-                if (v) {
-                    newVal.push(v);
+                const otherValue = otherField.get(otherEntity);
+                if (otherValue) {
+                    if (otherValue instanceof Array) {
+                        // avoid nested array if otherField is x2many too
+                        for (const v of otherValue) {
+                            newVal.push(v);
+                        }
```

Comment:
nested array is desirable because flatten list may be the same but the inner lists may have change so it should invoke computes. However, relational fields do not support nested fields at the moment, and this would be handy for related on `x2many.x2many` like here. Don't know how hard it is to make support of that though... Could keep flatten in the meantime, but let's but comment as a FIXME because this could have unintended behaviour like I explained here

Reply:
task-2261221